var/spack/repos/builtin/packages/cntk/package.py

```diff
@@ -49,6 +54,26 @@ class Cntk(Package):
     # Patch to change behaviour of lock file - https://github.com/Microsoft/CNTK/issues/62
     patch('lock-file.patch')
 
+    # It seems that cntk, at least version 2.0, can not be built with GCC
+    # beyond 4.8.5.
+    conflicts('%gcc@5:')
+
+    def patch(self):
```

Comment:
These changes don't seem related to this PR. If you found and fixed a build issue with `cntk`, that's great, but it could go in a different PR. This PR should only include changes to OpenCV and changes to other packages that result from that.